src/main/scala/firrtl/StabilizeModuleNames.scala

```diff
@@ -124,33 +221,26 @@ class StabilizeModuleNames extends Transform
     }
 
     val nameMap = moduleNameAnnos.groupBy(_.desiredName).mapValues { annos =>
-      annos.distinct.map { a =>
-        val referringModule = Target.referringModule(a.target).module
+      annos.map(a => Target.referringModule(a.target).module).distinct.map { referringModule =>
         require(modMap.contains(referringModule), "ModuleNameAnnotations may not refer to blackboxes")
         modMap(referringModule)
       }
     }
 
-    val strategies: Seq[Strategy] = Seq(
-      StabilizeModuleNames.exactName,
-      StabilizeModuleNames.portStructureName,
-      StabilizeModuleNames.contentsStructureName,
-      StabilizeModuleNames.contentsName,
+    val strategies = Seq(
+      ExactNamingStrategy,
```

Comment:
is there a strategy that corresponds to the current behavior (incrementing numerical suffixes), or is that just what you get if you don't apply this transform at all?

Reply:
That's what you get if you don't apply the transform/don't emit any module name annotations. the incrementing comes from chisel.